mm2src/coins/nft.rs

```diff
@@ -166,53 +245,284 @@ pub async fn update_nft(ctx: MmArc, req: UpdateNftReq) -> MmResult<(), UpdateNft
     nft_db.common.token_uri = token_uri;
+    nft_db.common.token_domain = token_domain;
     nft_db.common.metadata = moralis_meta.common.metadata;
     nft_db.common.last_token_uri_sync = moralis_meta.common.last_token_uri_sync;
     nft_db.common.last_metadata_sync = moralis_meta.common.last_metadata_sync;
     nft_db.common.possible_spam = moralis_meta.common.possible_spam;
     nft_db.uri_meta = uri_meta;
-    drop_mutability!(nft_db);
+    refresh_possible_spam(&storage, &req.chain, &mut nft_db, &req.url_antispam).await?;
+    refresh_possible_phishing(&storage, &req.chain, domains, &mut nft_db, &req.url_antispam).await?;
     storage
         .refresh_nft_metadata(&moralis_meta.chain, nft_db.clone())
         .await?;
-    let transfer_meta = TransferMeta::from(nft_db.clone());
+    update_transfer_meta_using_nft(&storage, &req.chain, &mut nft_db).await?;
+    Ok(())
+}
+
+/// The `update_transfer_meta_using_nft` function updates the transfer metadata associated with the given NFT.
+/// If metadata info contains potential spam links, function sets `possible_spam` true.
+async fn update_transfer_meta_using_nft<T>(storage: &T, chain: &Chain, nft: &mut Nft) -> MmResult<(), UpdateNftError>
+where
+    T: NftListStorageOps + NftTransferHistoryStorageOps,
+{
+    let transfer_meta = TransferMeta::from(nft.clone());
     storage
-        .update_transfers_meta_by_token_addr_id(&nft_db.chain, transfer_meta)
+        .update_transfers_meta_by_token_addr_id(chain, transfer_meta, nft.common.possible_spam)
         .await?;
     Ok(())
 }
 
-async fn get_moralis_nft_list(ctx: &MmArc, chain: &Chain, url: &Url) -> MmResult<Vec<Nft>, GetNftInfoError> {
+/// Extracts domains from uri_meta and token_domain.
+fn gather_domains(token_domain: &Option<String>, uri_meta: &UriMeta) -> HashSet<String> {
+    let mut domains = HashSet::new();
+    if let Some(domain) = token_domain {
+        domains.insert(domain.clone());
+    }
+    if let Some(domain) = &uri_meta.image_domain {
+        domains.insert(domain.clone());
+    }
+    if let Some(domain) = &uri_meta.animation_domain {
+        domains.insert(domain.clone());
+    }
+    if let Some(domain) = &uri_meta.external_domain {
+        domains.insert(domain.clone());
+    }
+    domains
+}
+
+/// Refreshes the `possible_spam` flag based on spam results.
+async fn refresh_possible_spam<T>(
+    storage: &T,
+    chain: &Chain,
+    nft_db: &mut Nft,
+    url_antispam: &Url,
+) -> MmResult<(), UpdateNftError>
+where
+    T: NftListStorageOps + NftTransferHistoryStorageOps,
+{
+    let address_hex = eth_addr_to_hex(&nft_db.common.token_address);
+    let spam_res = send_spam_request(chain, url_antispam, address_hex.clone()).await?;
+    if let Some(true) = spam_res.result.get(&nft_db.common.token_address) {
+        nft_db.common.possible_spam = true;
+        storage
+            .update_transfer_spam_by_token_address(chain, address_hex, true)
+            .await?;
```

Comment:
This call is not needed if `possible_spam` was `true` before, also can `possible_spam` change from `true` to `false`? meaning that an NFT that was flagged as possible spam passes some criteria and gets reconsidered as not being a spam.

Reply:
> This call is not needed if possible_spam was true before,

oh, yeah agree in `refresh_possible_spam` there is no need to send spam request again if possible spam is already true bcz of `nft_db.common.possible_spam = moralis_meta.common.possible_spam;`

> also can possible_spam change from true to false

Well we dont know how moralis marks their nft spam or not spam. 
But yeah it could be a situation that they changed `possible_spam` `from` true to `false`
But since we use separate db with spamy addresses I suggest to mark nft as spam if contact address has true in our db.

Another but :D I got the idea. lets check in `refresh_possible_spam` function whether the `spam` value for contract address  has changed in our database or not. If moralis returns false and our spam api returns false for contract in `refresh` functionality then lets leave this as `false`.

Reply:
> But yeah it could be a situation that they changed possible_spam from true to false

> But since we use separate db with spamy addresses I suggest to mark nft as spam if contact address has true in our db.

I consider this an edge case, but NFT collections can gain traction after sometime and some of the false positive possible spam NFTs due to the criteria used to identify spams can become popular in the future. I see that we don't change `true` to `false` anywhere, we can keep it this way for now until we get real cases from users feedback.